wallet-api/src/Language/PlutusTx/Interval.hs

```diff
@@ -0,0 +1,149 @@
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE DerivingStrategies   #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DeriveGeneric        #-}
+{-# LANGUAGE TemplateHaskell      #-}
+{-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE MonoLocalBinds       #-}
+-- | A type for intervals and associated functions.
+module Language.PlutusTx.Interval(
+      Slot(..)
+    , Interval(..)
+    , SlotRange
+    , interval
+    , from
+    , to
+    , singleton
+    , empty
+    , always
+    , member
+    , width
+    , contains
+    , before
+    , after
+    ) where
+
+import           Codec.Serialise.Class                    (Serialise)
+import           Data.Aeson                               (FromJSON, ToJSON)
+import           Data.Swagger.Internal.Schema             (ToSchema)
+import           GHC.Generics                             (Generic)
+import           Language.Haskell.TH
+
+import           Language.PlutusTx.Lift                   (makeLift)
+
+-- | Slot number
+newtype Slot = Slot { getSlot :: Int }
+    deriving (Eq, Ord, Show, Enum)
+    deriving stock (Generic)
+    deriving anyclass (ToSchema, FromJSON, ToJSON)
+    deriving newtype (Num, Real, Integral, Serialise)
+
+makeLift ''Slot
+
+-- | A possibly open-ended inclusive-exclusive interval of slots.
```

Comment:
I think "inclusive-exclusive" means "closed below open above", but you could write this out more explicitly, and maybe include a couple of examples.